src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -315,7 +317,12 @@ public void onMessage(String channel, String message) {
 
         } catch (JedisException e) {
 
-          if (running.get()) {
+          try {
+            lock.unlock();
+          }catch (IllegalMonitorStateException ex){
+            log.debug("lock is not held by MasterListener");
+          }
+          if (!isShutdown.get()) {
```

Comment:
Why do you need to check this? We had the lock until line 321 so obviously the isShutdown was false. And anyway by the the time you check it on line 325 to the log on line 326 it can be set to true.